linker/Linker.Steps/MarkStep.cs

```diff
@@ -1587,5 +1598,116 @@ protected virtual void MarkInterfaceImplementation (TypeDefinition type, Interfa
 			MarkCustomAttributes (iface);
 			MarkType (iface.InterfaceType);
 		}
+
+
+		void MarkSomethingUsedViaReflection (string reflectionMethod, Action<Collection<Instruction>, string, TypeDefinition, BindingFlags> markMethod, Collection<Instruction> instructions)
+		{
+			for (var i = 0; i < instructions.Count; i++) {
+				var instruction = instructions [i];
+				if (instruction.OpCode == OpCodes.Call || instruction.OpCode == OpCodes.Callvirt) {
+					var methodBeingCalled = instruction.Operand as MethodReference;
+					if (methodBeingCalled != null && methodBeingCalled.DeclaringType.Name == "Type" && methodBeingCalled.DeclaringType.Namespace == "System")
+					{
+						if (methodBeingCalled.Name == reflectionMethod) {
+							_context.Tracer.Push ($"Reflection-{methodBeingCalled}");
+							var nameOfThingUsedViaReflection = OperandOfNearestInstructionBefore<string> (i, OpCodes.Ldstr, instructions);
+							var bindingFlags = (BindingFlags) OperandOfNearestInstructionBefore<sbyte> (i, OpCodes.Ldc_I4_S, instructions);
+
+							// There might be more than one ldtoken opcode above the call in the IL stream. Be conservative and check all of
+							// the types which were loaded for the method being used.
+							var declaringTypesOfThingInvokedViaReflection = OperandsOfInstructionsBefore<TypeReference> (i, OpCodes.Ldtoken, instructions);
+							foreach (var declaringTypeOfThingInvokedViaReflection in declaringTypesOfThingInvokedViaReflection) {
+								var typeDefinition = declaringTypeOfThingInvokedViaReflection?.Resolve ();
+								if (typeDefinition != null)
+									markMethod (instructions, nameOfThingUsedViaReflection, typeDefinition, bindingFlags);
+							}
+							_context.Tracer.Pop ();
+						}
+					}
+				}
+			}
+		}
+
+		void MarkConstructorUsedViaReflection (Collection<Instruction> instructions, string unused1, TypeDefinition declaringTypeOfConstructorInvokedViaReflection, BindingFlags bindingFlags)
```

Comment:
* It marks more than one
* Repeating the name of method in parameters is not the convention we use

Reply:
> It marks more than one
I'll rename this method to be `MarkConstructorsUsedViaReflection`

> Repeating the name of method in parameters is not the convention we use
I'm unsure what you mean by this, sorry. Are you asking me to rename `declaringTypeOfConstructorInvokedViaReflection`?

Reply:
yes `declaringTypeOfConstructorInvokedViaReflection` is just awfully long and it repeats already know information

Reply:
Ok, I'll rename this parameter.